File: 557-ReverseWordsInAStringIii/557-ReverseWordsInAStringIii.py
# Last updated: 6/25/2026, 8:37:19 AM
class Solution:
    def reverse(self, charArr: List[str], start: int , end: int):
        startPtr = start
        endPtr = end
        while(startPtr < endPtr):
            t = charArr[startPtr]
            charArr[startPtr] = charArr[endPtr]
            charArr[endPtr] = t
            startPtr += 1
            endPtr -= 1
            
    def reverseWords(self, s: str) -> str:
        charArr = [letter for letter in s]
        p = 0
        for i in range(len(charArr)):
            if(charArr[i] == ' '):
                self.reverse(charArr, p, i - 1)
                p = i+1
        self.reverse(charArr, p, len(charArr)-1)
        return ("".join(charArr))
            
            
        

File: 557-ReverseWordsInAStringIii/557-ReverseWordsInAStringIii.java
// Last updated: 6/25/2026, 8:37:14 AM
class Solution {
    public String reverseWord(String word){
        char[] wordArr = word.toCharArray();
        int start = 0;
        int end = wordArr.length - 1;
        while(start < end){
            char temp = wordArr[start];
            wordArr[start] = wordArr[end];
            wordArr[end] = temp;
            start += 1;
            end -= 1;
        }
        String result = "";
        for(char t: wordArr){
            result += t;
        }
        return result;
    }
    public String reverseWords(String s) {
        String[] wordsArr = s.split(" ");
        int count = 0;
        for(String word: wordsArr){
            wordsArr[count] = reverseWord(word);
            count += 1;
        }
        return String.join(" ", wordsArr);
    }
}